app/models/report.rb

```diff
@@ -35,8 +38,13 @@ def report_type_abbr=(abbr)
   def run(*args)
     report_service = report_type.class_constant.new(*args)
     report_service.run
+    report_service
+  end
```

Comment:
I like returning self, argument here to ensure all the `base_<whatever>` reports and child classes return self in `def run`?

Reply:
nope - we need to return the 'report service' here (which is the derived report class). This whole structure is backwards (starting from the poor design making Report a model_